### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library and leverage on latest features of ExDoc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,9 +22,12 @@ erl_crash.dump
...
 kaffy-*.tar
 
-# Ignore local node_modules
+# Temporary files, for example, from tests.
+/tmp/
+
+# Ignore local node_modules.
 /node_modules/
 
-.elixir_ls
```
Comment (aesmail) [inline .gitignore:28]: @kianmeng  why are we keeping track of the `.elixir_ls` folder?

**Comment by aesmail** [comment]
Amazing effort @kianmeng 👏🏼
This will be extremely useful.
I just have a few points to clarify before merging this. Please check the comments on the files you've changed.

File: .gitignore (hunk 1 above)
Comment (kianmeng) [reply .gitignore:28]: > @kianmeng why are we keeping track of the `.elixir_ls` folder?

@aesmail Keeping track or removing? If it's the latter, ElixirLS already ignored `.elixir_ls` folder.

**Comment by kianmeng** [comment]
@aesmail Should I resolve the conflict and resubmit the PR again?

**Comment by aesmail** [comment]
@kianmeng yes please 🙏🏼

Will merge once it's resolved

**Comment by kianmeng** [comment]
@aesmail Done. Please check.

**Comment by aesmail** [comment]
Hello @kianmeng 
Sorry for the double work, but I'm still very interested in merging this PR since I'm trying to close all currently pending PRs.
Are you able to resolve the conflicts one last time so this can be merged along with the others for the v0.9.3 release?

**Comment by kianmeng** [comment]
@aesmail Sure do. Please check again.

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**